src/index.tsx

```diff
@@ -510,10 +506,13 @@ export function useValidatedInput<T extends FormValidations>(
       id: getInputId(name, id),
       className: getClasses("input", attrs.className),
       defaultValue: serverFormInfo?.submittedFormData?.lastName,
-      onChange: callAll(onChange, (e: React.ChangeEvent<HTMLInputElement>) => {
-        setDirty(true);
-        setValue(e.target.value);
-      }),
+      onChange: (event: React.ChangeEvent<HTMLInputElement>) => {
+        onChange?.(event);
+        if (!event.defaultPrevented) {
+          setDirty(true);
+          setValue(event.target.value);
```

Comment:
I'm not sure this is what we want if the `change` event is prevented - since that happens _after_ the input value has been set in the DOM (at least for text and similar inputs - would need to look closer on radio/checkbox).  If we don't set the controlled value we can get into weird UI states:

<img width="454" alt="Screen Shot 2022-09-29 at 8 43 07 AM" src="https://user-images.githubusercontent.com/1609022/193035464-4eec5599-1b22-45b3-a99a-44981ae1d68a.png">

Users can use `onKeyDown` currently to prevent typed characters.

I've been thinking for a while too that I'd like to take a stab at making this entire library uncontrolled and just always use `inputEl.value`, which would remove this `setValue` entirely.  I _think_ that's feasible and would get rid of the `useState` so might have the worlds tiniest perf improvement 🎻